### PR TITLE
Correct path/filename for server.jar for Fabric

### PR DESCRIPTION
**Pull request by JackGlobetrotter**
Fabric uses `fabric-server-launch.jar` to start the server. This file does not contain the version.json. Instead the `server.jar` containing this file is located under `versions/$VERSION/server-$VERSION.jar`. This PR corrects the path if Fabric is detected to read the correct file.

### CODE REVIEW

**Comment by A-wels** [comment]
Do you know if this is only the case for fabric servers?
I think it could be a good idea to try and read the version as before and, on error, try the other approach.

This may allow for a broader fix.
What do you think @JackGlobetrotter @gekigek99

**Comment by JackGlobetrotter** [comment]
I am not so familliar with other servers than fabric... to do this in "error handeling" might be a better solution yes!

**Comment by JackGlobetrotter** [comment]
@A-wels made some changes to preserve default methode and failover to default name only if there is an error, what do you think?

File: lib/config/config-utils.go [hunk 1]
```diff
@@ -198,6 +203,10 @@ func (c *Configuration) getVersionInfo() (string, int, *errco.MshLog) {
 		return utility.FirstNon("", info.Version1, info.Version2), info.Protocol, nil
 	}
 
+	if c.Server.FileName != "server.jar" { //check if it matches the deafult name, if not
+		return c.getVersionInfo("server.jar") //fallback to default naming scheme
+	}
+
```
Comment (gekigek99) [inline lib/config/config-utils.go:209]: @JackGlobetrotter sorry for late reply, I'm super busy irl. Why did you add this check? it seems to me that it makes useless the logic implemented at begin of function.

File: lib/config/config-utils.go [hunk 2]
```diff
@@ -165,8 +165,13 @@ func (c *Configuration) loadIcon() *errco.MshLog {
 // In case of error "", -1, *errco.MshLog are returned.
 //
 // (checkout version.json info: https://minecraft.fandom.com/wiki/Version.json)
-func (c *Configuration) getVersionInfo() (string, int, *errco.MshLog) {
-	reader, err := zip.OpenReader(filepath.Join(c.Server.Folder, c.Server.FileName))
+func (c *Configuration) getVersionInfo(altServerName ...string) (string, int, *errco.MshLog) {
```
Comment (gekigek99) [inline lib/config/config-utils.go:168]: @JackGlobetrotter why `altServerName ...string` ? isn't it enough to have one string parameter?

Comment (JackGlobetrotter) [reply lib/config/config-utils.go:168]: @gekigek99 i wanted to make the parameter optional, so that the function could be used w/o params if needed elswhere in the futur? But could also be `altServerName string` if not needed

**Comment by gekigek99** [comment]
Thanks @JackGlobetrotter @A-wels for both the PR and explanation. 

Hopefully you'll excuse me for late accept but I was focusing on university in the last months.

**Comment by A-wels** [comment]
Don't say sorry for that. I know first hand that university can suck all time away.

**Comment by JackGlobetrotter** [comment]
Can relate to that, please don't be :) 
Thanks for accepting the PR.
